tests/test-naughty-legacy.lua

```diff
@@ -563,120 +548,86 @@ table.insert(steps, function()
     return true
 end)
 
--- Test more advanced features than what notify-send can provide.
-if has_cmd_send then
-
-    local cmd = { 'dbus-send',
-        '--type=method_call',
-        '--print-reply=literal',
-        '--dest=org.freedesktop.Notifications',
-        '/org/freedesktop/Notifications',
-        'org.freedesktop.Notifications.Notify',
-        'string:"Awesome test"',
-        'uint32:0',
-        'string:""',
-        'string:"title"',
-        'string:"message body"',
-        'array:string:1,one,2,two,3,three',
-        'dict:string:string:"",""',
-        'int32:25000'
-    }
-
-    -- Test the actions.
-    table.insert(steps, function()
+-- Test the actions.
+table.insert(steps, function()
 
-        assert(#active == 0)
+    assert(#active == 0)
 
-        spawn(cmd)
+    send_notify("Awesome test", 0, "", "title", "message body",
+        { "1", "one", "2", "two", "3", "three" }, {}, 25000)
 
-        return true
-    end)
+    return true
+end)
 
-    table.insert(steps, function()
-        if #active == 0 then return end
+table.insert(steps, function()
+    if #active == 0 then return end
 
-        assert(#active == 1)
-        local n = active[1]
+    assert(#active == 1)
+    local n = active[1]
 
-        assert(n.box)
-        assert(#n.actions == 3)
-        assert(n.actions[1].name == "one"  )
-        assert(n.actions[2].name == "two"  )
-        assert(n.actions[3].name == "three")
+    assert(n.box)
+    assert(#n.actions == 3)
+    assert(n.actions[1].name == "one"  )
+    assert(n.actions[2].name == "two"  )
+    assert(n.actions[3].name == "three")
 
-        n:destroy()
+    n:destroy()
 
-        return true
-    end)
+    return true
+end)
 
-    --TODO Test too many actions.
+--TODO Test too many actions.
 
-    --TODO Test action with long names.
+--TODO Test action with long names.
 
-    local nid, name_u, message_u, actions_u = nil
+local nid, name_u, message_u, actions_u = nil
 
-    -- Test updating a notification.
-    table.insert(steps, function()
-        spawn.easy_async(cmd, function(out)
-            nid = tonumber(out:match(" [0-9]+"):match("[0-9]+"))
+-- Test updating a notification.
```

Comment:
This should/could also test https://github.com/awesomeWM/awesome/issues/2678.
I assume it works here, because the id is assigned throught the callback?

Reply:
Uhm, there is already a test for that. Exactly this piece here. It gets the notification it and then replaces the notification and tests that it was really replaced. I did not look at #2678, but I would guess that it is a problem with the Lua API, not the DBus API.

Reply:
Yes, here it appears to use the id via the callback, while with #2678 it appears to not get returned?